File: Coding-Interviews/21-调整数组顺序使奇数位于偶数前面.cpp
#include <vector>
using namespace std;
class Solution
{
public:
    vector<int> exchange(vector<int>& nums)
    {
        int index = 0;
        for (int i = 0; i < nums.size(); i++) {
            if (nums[i] % 2 == 1) {
                swap(nums[index], nums[i]);
                index++;
            }
        }
        return nums;
    }
};

File: Coding-Interviews/21-调整数组顺序使奇数位于偶数前面.py
from typing import List


class Solution:
    def exchange(self, nums: List[int]) -> List[int]:
        # [0,k)为奇数
        k, i = 0, 0
        for i in range(len(nums)):
            if nums[i] % 2:
                nums[k], nums[i] = nums[i], nums[k]
                k += 1
        return nums
